Mathlib/FieldTheory/SeparableDegree.lean

```diff
@@ -268,31 +249,539 @@ def embProdEmbOfIsAlgebraic [Algebra E K] [IsScalarTower F E K] (halg : Algebra.
+    · simp only [h, _root_.map_eq_zero, exists_eq]
+    have : F⟮f.coeff n⟯ ≤ E' := adjoin.mono _ _ _ <|
+      Set.singleton_subset_iff.2 <| f.coeff_mem_frange n h
+    exact ⟨⟨f.coeff n, this <| mem_adjoin_simple_self F (f.coeff n)⟩, rfl⟩
+  obtain ⟨g, h⟩ := f.mem_lifts.1 h
+  have hx : x ∈ restrictScalars F E'⟮x⟯ := mem_adjoin_simple_self _ x
+  have hzero : aeval x g = 0 := by
+    simpa only [← h, aeval_map_algebraMap] using minpoly.aeval E x
+  have halg : IsIntegral E' x := (IsSeparable.isAlgebraic F E).trans
+    (IsSeparable.isAlgebraic E K) x |>.isIntegral.tower_top
+  have hsep : f.Separable := IsSeparable.separable E x
+  rw [← h, separable_map] at hsep
+  replace hsep := hsep.of_dvd <| minpoly.dvd _ _ hzero
+  haveI : IsSeparable F E' := isSeparable_tower_bot_of_isSeparable F E' E
+  haveI := (isSeparable_adjoin_simple_iff_separable _ _).2 hsep
+  haveI := adjoin.finiteDimensional halg
+  letI : Algebra E' E'⟮x⟯ := Subalgebra.algebra E'⟮x⟯.toSubalgebra
+  letI : Module E' E'⟮x⟯ := Algebra.toModule
+  letI : SMul E' E'⟮x⟯ := Algebra.toSMul
+  haveI : IsScalarTower F E' E'⟮x⟯ := IntermediateField.isScalarTower E'⟮x⟯
+  haveI : FiniteDimensional F E'⟮x⟯ := FiniteDimensional.trans F E' E'⟮x⟯
+  have := finSepDegree_mul_finSepDegree_of_isAlgebraic F E' E'⟮x⟯ (IsSeparable.isAlgebraic _ _)
+  rw [finSepDegree_eq_finrank_of_isSeparable F E',
+    finSepDegree_eq_finrank_of_isSeparable E' E'⟮x⟯,
+    FiniteDimensional.finrank_mul_finrank F E' E'⟮x⟯,
+    eq_comm, finSepDegree_eq_finrank_iff F E'⟮x⟯] at this
+  change IsSeparable F (restrictScalars F E'⟮x⟯) at this
+  exact separable_of_mem_isSeparable F K hx
+
+namespace Field
+
+/-- If `x` and `y` are both separable elements, then `x * y` is also a separable element. -/
+theorem separable_mul {x y : E} (hx : (minpoly F x).Separable) (hy : (minpoly F y).Separable) :
+    (minpoly F (x * y)).Separable :=
+  haveI := isSeparable_adjoin_two_of_separable F E hx hy
+  separable_of_mem_isSeparable F E <| F⟮x, y⟯.mul_mem (subset_adjoin F _ (Set.mem_insert x {y}))
+    (subset_adjoin F _ (Set.mem_insert_of_mem x rfl))
+
+/-- If `x` and `y` are both separable elements, then `x + y` is also a separable element. -/
+theorem separable_add {x y : E} (hx : (minpoly F x).Separable) (hy : (minpoly F y).Separable) :
+    (minpoly F (x + y)).Separable :=
+  haveI := isSeparable_adjoin_two_of_separable F E hx hy
+  separable_of_mem_isSeparable F E <| F⟮x, y⟯.add_mem (subset_adjoin F _ (Set.mem_insert x {y}))
+    (subset_adjoin F _ (Set.mem_insert_of_mem x rfl))
+
+/-- Any element `x` of `F` is a separable element of `E / F` when embedded into `E`. -/
+theorem separable_algebraMap (x : F) : (minpoly F ((algebraMap F E) x)).Separable := by
+  rw [minpoly.algebraMap_eq (algebraMap F E).injective]
+  exact IsSeparable.separable F x
+
+/-- If `x` is a separable element, then `x⁻¹` is also a separable element. -/
+theorem separable_inv (x : E) (hx : (minpoly F x).Separable) : (minpoly F x⁻¹).Separable :=
```

Comment:
Maybe it is worth to introduce the... subfield/intermediate field/whatever of separable elements. What do you think?

Reply:
It's [here](https://github.com/leanprover-community/mathlib4/pull/8696/files#diff-fbae6aa6a45b5781c17dc0397d1edfbcf1ed608887c0c6aa87cf2f0a57d44145R86-R96) in another file so you just need to wait for another PR :)

Reply:
> It's [here](https://github.com/leanprover-community/mathlib4/pull/8696/files#diff-fbae6aa6a45b5781c17dc0397d1edfbcf1ed608887c0c6aa87cf2f0a57d44145R86-R96) in a file untouched by this PR so you just need to wait for another PR :)

Oh but that definition makes use of these `separable_*` results XD

Reply:
Yeah, hence this PR is prerequisite to the follow-up PR :)